### DISCUSSION TITLE
Show image preview and thumbnails for shared files and directories

### DISCUSSION THREAD

**Discussion by nook24** [filebrowser/filebrowser#1271] [Ideas]
**Is your feature request related to a problem? Please describe.**
When sharing files or folders for external users they always have to download the images first before they can watch them. The Sharing Site has no preview, only a download button.

![single_file](https://user-images.githubusercontent.com/9019992/106104217-6d096480-6142-11eb-8309-b625ab6f94a4.png)

The same for folders. #1141
![folder](https://user-images.githubusercontent.com/9019992/106104213-6b3fa100-6142-11eb-84c3-08d26d44710d.png)

**Describe the solution you'd like**
Files:
If its a picture, it would be good to open shared pictures with a large preview.

Folders:
For shared folders it would be good to get a thumbnails for all files.

**Comment by devtechk** [comment]
Hi! I'm looking for the same feature I mean when you share a file or a folder give the user a preview before download it.
I'm planning to edit the code (out of my job) to support this feature, if anyone wants to help me I'll be gratefull!
